reana_server/reana_admin.py

```diff
@@ -214,39 +209,102 @@ def import_users(ctx, admin_access_token, file_):
+                         ' active access token.')
+        elif (not force and user.access_token_status in
+              [UserTokenStatus.revoked.name, None]):
+            error_msg = (f'User {user.id_} ({user.email}) access token status'
+                         f' is {user.access_token_status}, if you want to'
+                         ' proceed in any case use --force option.')
+        if error_msg:
+            click.secho(f'ERROR: {error_msg}', fg='red')
+            sys.exit(1)
+
+        user_granted_token = secrets.token_urlsafe(16)
+        user.access_token = user_granted_token
+        Session.commit()
+        log_msg = (f'Token for user {user.id_} ({user.email}) granted.\n'
+                   f'\nToken: {user_granted_token}')
+        click.secho(log_msg, fg='green')
+        admin.log_action(AuditLogAction.grant_token, {'reana_admin': log_msg})
+        # send notification to user by email
+        email_subject = 'REANA access token granted'
+        email_body = f'Dear {user.full_name},\n\nYour REANA access token has' \
+            f' been granted, please find it on https://{REANA_URL}/profile' \
+            '\n\nThe REANA support team'
+        send_email(user.email, email_subject, email_body)
+
+    except Exception as e:
+        click.secho(
+            'Something went wrong while granting token:\n{}'.format(e),
+            fg='red', err=True)
 
 
 @reana_admin.command(
     'token-revoke', help='Revoke selected user\'s token.')
+@admin_access_token_option
 @click.option(
     '--id',
+    'id_',
     help='The id of the user.')
 @click.option(
     '-e',
     '--email',
     help='The email of the user.')
-@admin_access_token_option
-def token_revoke(id, email, admin_access_token):
-    """."""
-    user_id = 'dc4aef21-c324-4fd5-8480-c724de1f0c06'
-    user_email = 'john.doe@google.com'
-    # deactivate REANA token in UserToken table
-    # create an audit log entry
-    # send notification to user by email
-    click.secho(f'User\'s {user_id} ({user_email}) token successfully revoked',
-                fg='green')
+def token_revoke(admin_access_token, id_, email):
+    """Revoke selected user's token."""
+    try:
+        admin = User.query.filter_by(id_=ADMIN_USER_ID).one_or_none()
+        if admin_access_token != admin.access_token:
+            raise ValueError('Admin access token invalid.')
+        user = _get_user_by_criteria(id_, email)
+
+        error_msg = None
+        if not user:
+            error_msg = f'User {id_ or email} does not exist.'
+        elif not user.access_token:
+            error_msg = (f'User {user.id_} ({user.email}) does not have an'
+                         ' active access token.')
+        if error_msg:
+            click.secho(f'ERROR: {error_msg}', fg='red')
+            sys.exit(1)
+
+        user.active_token.status = UserTokenStatus.revoked
+        Session.commit()
+        log_msg = f'User\'s {user.id_} ({user.email}) token successfully' \
+            ' revoked.'
```

Comment:
Messages to be amended according to https://github.com/reanahub/docs.reana.io/pull/44#discussion_r417879081.